pilot/pkg/networking/core/v1alpha3/extension/wasmplugin.go

```diff
@@ -52,6 +52,29 @@ func PopAppend(list []*hcm.HttpFilter,
 	return list
 }
 
+// popAppendNetwork takes a list of filters and a set of WASM plugins, keyed by phase. It will remove all
+// plugins of a provided phase from the WASM plugin set and append them to the list of filters
+func popAppendNetwork(list []*listener.Filter,
+	filterMap map[extensions.PluginPhase][]*model.WasmPluginWrapper,
+	phase extensions.PluginPhase,
+) []*listener.Filter {
+	for _, ext := range filterMap[phase] {
+		list = append(list, toEnvoyNetworkFilter(ext))
+	}
+	delete(filterMap, phase)
+	return list
+}
+
+// BuildNetworkFilters builds network WASM filters from a set of WASM plugins, orderded by phase.
+func BuildNetworkFilters(filterMap map[extensions.PluginPhase][]*model.WasmPluginWrapper) []*listener.Filter {
+	filters := make([]*listener.Filter, 0)
+	filters = popAppendNetwork(filters, filterMap, extensions.PluginPhase_AUTHN)
+	filters = popAppendNetwork(filters, filterMap, extensions.PluginPhase_AUTHZ)
+	filters = popAppendNetwork(filters, filterMap, extensions.PluginPhase_STATS)
+	filters = popAppendNetwork(filters, filterMap, extensions.PluginPhase_UNSPECIFIED_PHASE)
```

Comment:
this seems broken since we do not insert the Istio authn, authz, and stats filters in between them

Reply:
yeah. This is removed now.